index.js

```diff
@@ -1061,12 +1061,12 @@ Jimp.prototype.autocrop = function() {
  * @returns this for chaining of methods
 */
 Jimp.prototype.blit = function (src, x, y, srcx, srcy, srcw, srch, cb) {
-    if ("object" != typeof src || src.constructor != Jimp)
+    if (!(src instanceof Jimp))
         return throwError.call(this, "The source must be a Jimp image", cb);
     if ("number" != typeof x || "number" != typeof y)
         return throwError.call(this, "x and y must be numbers", cb);
 
-    if ("function" == typeof srcx) {
+    if (srcx instanceof Function) {
```

Comment:
I honestly couldn't think of a scenario where it makes a difference, but personally I would recommend to use typeof only for primitives.